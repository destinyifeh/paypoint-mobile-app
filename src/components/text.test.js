'use strict';

import React from 'react';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import Text from './text';

configure({
  adapter: new Adapter()
});

it('Bold text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      bold
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Failed status text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      isFailedStatus
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Pending status text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      isPendingStatus
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Success status text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      isSuccessStatus
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Status text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      isStatus
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Black text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      black
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Big text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      big
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Bigger text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      bigger
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Center text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      center
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Green text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      green
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Small text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      small
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Red text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      red
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Right-aligned text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      right
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Title text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      title
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Uppercase text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      uppercase
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('White text renders correctly', () => {
  const tree = renderer
    .create(<Text 
      white
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
