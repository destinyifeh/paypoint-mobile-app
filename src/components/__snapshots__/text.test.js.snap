// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big text renders correctly 1`] = `
<Text
  big={true}
  style={
    Object {
      "color": "#808593",
      "fontSize": 22,
      "lineHeight": 24,
    }
  }
/>
`;

exports[`Bigger text renders correctly 1`] = `
<Text
  bigger={true}
  style={
    Object {
      "color": "#808593",
      "fontSize": 26,
      "lineHeight": 32,
    }
  }
/>
`;

exports[`Black text renders correctly 1`] = `
<Text
  black={true}
  style={
    Object {
      "color": "#3A3B3B",
    }
  }
/>
`;

exports[`Bold text renders correctly 1`] = `
<Text
  bold={true}
  style={
    Object {
      "color": "#808593",
      "fontWeight": "bold",
    }
  }
/>
`;

exports[`Center text renders correctly 1`] = `
<Text
  center={true}
  style={
    Object {
      "color": "#808593",
      "textAlign": "center",
      "textAlignVertical": "center",
    }
  }
/>
`;

exports[`Failed status text renders correctly 1`] = `
<Text
  isFailedStatus={true}
  style={
    Object {
      "color": "#EE312A",
      "fontSize": 12,
    }
  }
/>
`;

exports[`Green text renders correctly 1`] = `
<Text
  green={true}
  style={
    Object {
      "color": "#32BE69",
    }
  }
/>
`;

exports[`Pending status text renders correctly 1`] = `
<Text
  isPendingStatus={true}
  style={
    Object {
      "color": "#EABB73",
      "fontSize": 12,
    }
  }
/>
`;

exports[`Red text renders correctly 1`] = `
<Text
  red={true}
  style={
    Object {
      "color": "#EE312A",
    }
  }
/>
`;

exports[`Right-aligned text renders correctly 1`] = `
<Text
  right={true}
  style={
    Object {
      "color": "#808593",
      "textAlign": "right",
    }
  }
/>
`;

exports[`Small text renders correctly 1`] = `
<Text
  small={true}
  style={
    Object {
      "color": "#808593",
      "fontSize": 12,
      "lineHeight": undefined,
    }
  }
/>
`;

exports[`Status text renders correctly 1`] = `
<Text
  isStatus={true}
  style={
    Object {
      "color": "#808593",
      "fontSize": 12,
    }
  }
/>
`;

exports[`Success status text renders correctly 1`] = `
<Text
  isSuccessStatus={true}
  style={
    Object {
      "color": "#32BE69",
      "fontSize": 12,
    }
  }
/>
`;

exports[`Title text renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#808593",
      "fontSize": 16,
    }
  }
  title={true}
/>
`;

exports[`Uppercase text renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#808593",
      "textTransform": "uppercase",
    }
  }
  uppercase={true}
/>
`;

exports[`White text renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
    }
  }
  white={true}
/>
`;
